src/ngraph/runtime/executable.hpp

```diff
@@ -112,4 +116,6 @@ class ngraph::runtime::Executable
 private:
     ngraph::ParameterVector m_parameters;
     ngraph::ResultVector m_results;
+    // default pipeline_depth of 2, specific executable can override.
+    size_t m_default_pipeline_depth = 2;
```

Comment:
Remove this. Put the value inside get_default_pipeline_depth() implementation. Because you made m_default_pipeline_depth private it can't be changed by a derived class anyway.